src/controllers/movie.js

```diff
@@ -0,0 +1,67 @@
+const Movie = require("../models/movies");
+const { setCache, getCache } = require("../configs/cacheConfig");
+const Rating = require("../models/ratings");
+
+const getAllMovies = async (req, res) => {
+  try {
+    let movies = await Movie.findAll();
+    if (movies.length == 0) {
+      return res.status(404).json({ message: "No Result Found" });
+    }
+    return res.status(200).json(movies);
+  } catch (error) {
+    res.status(500).json({ message: "Something Went Wrong!" });
+  }
+};
+
+const getMovieById = async (req, res) => {
+  try {
+    let { movieId } = req.params;
+    let cachedData = await getCache(`movie_${movieId}`);
+    if (cachedData) {
+      return res
+        .status(200)
+        .json({ movieDetails: cachedData.movie, isCachedData: true });
+    }
+    let movie = await Movie.find({
+      where: {
+        id: movieId,
+      },
+    });
+    if (movie) {
```

Comment:
could have avoided if - else block by single if block. For example -:
if(!movie) { res.status(500) }